src/main/java/it/gov/pagopa/pu/bff/util/DocumentComposition.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    @Value
+    private static class BaseUriAndTemplateBody {
+        private String baseUri;
+        private String templateBody;
+    }
+
+    private static BaseUriAndTemplateBody preloadTemplate(TemplateType templateType) throws IOException {
+        log.debug("Start pre-loading template with templateType={}", templateType);
+
+        String templateResourceName = templateType.getHtmlTemplate();
+        URL templateUrl = getClasspathResourceURL(templateResourceName);
+        log.debug("Template with templateResourceName={} located at URL={}", templateResourceName, templateUrl);
+
+        String baseUri = templateUrl.toString().replaceFirst("/[^/]*$", "/");
+        String templateBody = loadTemplateBody(templateUrl);
+
+        log.debug("Template resources baseUri={}", baseUri);
+        return new BaseUriAndTemplateBody(baseUri, templateBody);
+    }
+
+    private static String loadTemplateBody(URL templateUrl) throws IOException {
+
+        String templateContent;
+        try (InputStream templateIn = templateUrl.openStream()) {
+            templateContent = StreamUtils.copyToString(templateIn, StandardCharsets.UTF_8);
+        } catch (IOException exc) {
+            log.error("Loading Document Composition Template " + templateUrl, exc);
+            throw exc;
+        }
+        return templateContent;
+    }
+
+    private static URL getClasspathResourceURL(String resourceName) {
+        return Thread.currentThread().getContextClassLoader().getResource(resourceName);
+    }
+
+    public String executeTextTemplate(TemplateType templateType, Object model) throws
+            IOException, TemplateException {
+        log.info("Execute templateType={} START", templateType);
+        StringWriter stringWriter = new StringWriter();
+
+        Template template = freemarker.getTemplate(templateType.name());
+        template.process(model, stringWriter);
+
+        log.info("Execute templateType={} END", templateType);
+        return stringWriter.getBuffer().toString();
+    }
+
+    public byte[] executePdfTemplate(TemplateType templateType, Object model) throws IOException, TemplateException {
+        String html = executeTextTemplate(templateType, model);
+
+        String baseUri = baseUris.get(templateType);
+        log.info("Pdf conversion start for templateType={} with baseUri={}", templateType, baseUri);
+
+        byte[] pdf = html2Pdf(baseUri, html);
+
+        log.info("Pdf conversion done");
+        return pdf;
+    }
+
+    private byte[] html2Pdf(String baseUri, String html) throws IOException {
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        PdfRendererBuilder builder = new PdfRendererBuilder();
+        builder.usePdfUaAccessbility(true);
+        builder.usePdfAConformance(PdfRendererBuilder.PdfAConformance.PDFA_3_A);
+        builder.withHtmlContent(html, baseUri);
+        builder.toStream(baos);
+        builder.run();
+        baos.close();
```

Comment:
use try with resource instead, otherwise you will no close the stream if an exception occurs